pycbc/io/hdf.py

```diff
@@ -1090,20 +1086,18 @@ def to_coinc_hdf_object(self, file_name):
                 vals = vals_valid[ifo][0]
                 valid = vals_valid[ifo][1]
                 vals[np.logical_not(valid)] = -1.
-                ofd.create_dataset(ifo + '_'+field, data=vals,
-                                   dtype=np.float32)
+                ofd[f'{ifo}_{field}'] = vals
```

Comment:
maybe good to put a line of comment before the 'vals_valid' lines above to explain what this is intended to do?